Utils/FaviconFetcher.swift

```diff
@@ -93,73 +94,84 @@ public class FaviconFetcher : NSObject, NSXMLParserDelegate {
-                    return self.parseHTMLForFavicons(url)
-                }
+            let element = SWXMLHash.lazy(data)
+            var reloadUrl: NSURL? = nil
 
-                var bestType = IconType.NoneFound
-                element.iterateWithRootXPath("//head//link[contains(@rel, 'icon')]") { link in
-                    var iconType: IconType? = nil
-                    if let rel = link.attribute("rel") {
-                        switch (rel) {
-                            case "shortcut icon":
-                                iconType = .Icon
-                            case "icon":
-                                iconType = .Icon
-                            case "apple-touch-icon":
-                                iconType = .AppleIcon
-                            case "apple-touch-icon-precomposed":
-                                iconType = .AppleIconPrecomposed
-                            default:
-                                iconType = nil
-                        }
+            //lets see if this page requires a refresh to get the actual page
+            do {
+                let refreshNode = try element["html"]["head"]["meta"].withAttr("http-equiv", "Refresh").element
+                if let urlParam = refreshNode?.attributes["content"],
+                    let index = urlParam.rangeOfString("URL="),
+                    let url = NSURL(string: urlParam.substringFromIndex(index.startIndex.advancedBy(4))) {
+                        reloadUrl = url
                     }
+            } catch {}
 
-                    guard let href = link.attribute("href") where iconType != nil else {
-                        return
-                    }
 
-                    if (href.endsWith(".ico")) {
-                        iconType = .Guess
-                    }
+            if let url = reloadUrl {
+              return self.parseHTMLForFavicons(url)
+            }
 
-                    if let type = iconType where !bestType.isPreferredTo(type),
-                        let iconUrl = NSURL(string: href, relativeToURL: url) {
-                            let icon = Favicon(url: iconUrl.absoluteString, date: NSDate(), type: type)
-                            // If we already have a list of Favicons going already, then add it…
-                            if (type == bestType) {
-                                icons.append(icon)
-                            } else {
-                                // otherwise, this is the first in a new best yet type.
-                                icons = [icon]
-                                bestType = type
-                            }
+            //map all the icon types to an array of Favicons
+            let headElement = element["html"]["head"]["link"]
+            let relAttributes = ["shortcut icon","icon","apple-touch-icon","apple-touch-icon-precomposed"]
+            let iconsMapped = relAttributes.flatMap({ (relIcon:String) -> Favicon? in
```

Comment:
I went with a functional approach using flatmap. I dont know if this is okay for styling within the project. But it made sense to map the different images to an array of Favicons.